query-node/mappings/src/common.ts

```diff
@@ -0,0 +1,83 @@
+import BN from 'bn.js'
+import { SubstrateEvent } from '@dzlzv/hydra-common'
+import { DatabaseManager } from '@dzlzv/hydra-db-utils'
+import { u64 } from '@polkadot/types/primitive';
+
+// Asset
+import {
+  DataObjectOwner,
+  DataObject,
+  LiaisonJudgement,
+  Network,
+} from 'query-node'
+import {
+  ContentParameters
+} from '@joystream/types/augment'
+
+const currentNetwork = Network.BABYLON
+
+/*
+  Reports that insurmountable inconsistent state has been encountered and throws an exception.
+*/
+export function inconsistentState(extraInfo: string, data?: unknown): never {
+  const errorMessage = 'Inconsistent state: ' + extraInfo
+
+  // log error
+  logger.error(errorMessage, data)
+
+  // throw exception
+  throw errorMessage
+}
+
+/*
+  Prepares data object from content parameters.
+*/
+export async function prepareDataObject(
+  contentParameters: ContentParameters,
+  blockNumber: number,
+  owner: typeof DataObjectOwner,
+): Promise<DataObject> {
+  const dataObject = new DataObject({
+    owner,
+    createdInBlock: blockNumber,
+    typeId: contentParameters.type_id.toNumber(),
+    // `size` is masked by `size` special name in `Struct` so there needs to be `.get('size') as unknown as u64` to retrieve proper value
+    size: (contentParameters.get('size') as unknown as u64).toBn(),
+    liaisonJudgement: LiaisonJudgement.PENDING, // judgement is pending at start; liaison id is set when content is accepted/rejected
+    ipfsContentId: contentParameters.ipfs_content_id.toHex(),
```

Comment:
We expect ipfsContentId to be a Text, so don't convert to hex, instead convert to string.

Reply:
Conversion to text added via `.toString()`